### PR TITLE
Migration OCP 3 to OCP 4

### PR DESCRIPTION
**Pull request by mramendi**

### CODE REVIEW

File: doc-content/enterprise-only/openshift/migration-ocp3-4-bc-proc.adoc [hunk 1]
```diff
@@ -0,0 +1,47 @@
+[id='migration-ocp3-4-bc-proc-{context}']
+= Migrating information in {CENTRAL}
+
+If you have an existing authoring environment in {OPENSHIFT} version 3, you can copy the `.niogit` repository and the Maven repository from {CENTRAL} in this environment to {CENTRAL} in a new deployment on {OPENSHIFT} version 4. This action makes all the same projects and artifacts available in the new deployment.
+
+.Prerequisites
+
+* You must have a machine that has network access to both the {OPENSHIFT} version 3 and {OPENSHIFT} version 4 infrastructures.
+* The `oc` command-line client from {OPENSHIFT} version 4 must be installed on the machine.
```
Comment (ruromero) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-bc-proc.adoc:9]: Maybe link to the instructions to get the `oc` command? Just a suggestion

File: doc-content/enterprise-only/openshift/migration-ocp3-4-mysql-proc.adoc [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+oc get dc
+----
++
+Find the `mysql` deployment configuration that corresponds to the {KIE_SERVER}.
++
+. View the configuration YAML of the deployment configuration, for example:
++
+[subs="attributes,verbatim,macros,specialchars,replacements"]
+----
+oc edit dc/myapp-mysql
```
Comment (ruromero) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-mysql-proc.adoc:29]: If you want a oneliner for getting the values:
```
oc get dc/myapp-mysql -ojsonpath='{.spec.template.spec.containers[0].env[?(@.name=="MYSQL_USER")]}'.value
oc get dc/myapp-mysql -ojsonpath='{.spec.template.spec.containers[0].env[?(@.name=="MYSQL_PASSWORD")]}'.value
```

File: doc-content/enterprise-only/openshift/migration-ocp3-4-postgresql-proc.adoc [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
++
+----
+oc get dc
+----
++
+Find the `postgresql` deployment configuration that corresponds to the {KIE_SERVER}.
++
+. View the configuration YAML of the deployment configuration, for example:
++
+[subs="attributes,verbatim,macros,specialchars,replacements"]
```
Comment (ruromero) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-postgresql-proc.adoc:27]: You can mention the oneliner if you don't think it's too confusing
```
oc get dc/myapp-postgresql -ojsonpath='{.spec.template.spec.containers[0].env[?(@.name=="POSTGRESQL_USER")]}'.value
oc get dc/myapp-postgresql -ojsonpath='{.spec.template.spec.containers[0].env[?(@.name=="POSTGRESQL_PASSWORD")]}'.value
```

File: doc-content/enterprise-only/openshift/migration-ocp3-4-postgresql-proc.adoc [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+. Restore the database dump using the recorded username and database name, for example:
++
+[subs="attributes,verbatim,macros,specialchars,replacements"]
+----
+psql -h 127.0.0.1 -d {PRODUCT_INIT}7 -U {PRODUCT_INIT} -W <pgdump.sql 
+----
++
+When prompted, enter the recorded password. The restoration can take considerable time. 
++
+Ignore database error messages. These messages are displayed because a database with the sane name already exists.
```
Comment (ruromero) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-postgresql-proc.adoc:111]: I wouldn't say just _ignore_ any error message but something like _review the error messages and disregard those mentioning the object already exist_

File: doc-content/enterprise-only/openshift/migration-ocp3-4-bc-proc.adoc [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+.Prerequisites
+
+* You must have a machine that has network access to both the {OPENSHIFT} version 3 and {OPENSHIFT} version 4 infrastructures.
+* The `oc` command-line client from {OPENSHIFT} version 4 must be installed on the machine. For instructions about installing the command-line client, see https://access.redhat.com/documentation/en-us/openshift_container_platform/4.4/html/cli_tools/openshift-cli-oc#installing-the-cli[_CLI tools_] in {OPENSHIFT} documentation.
+
+.Procedure
+
+. Ensure that no web clients and no client applications are connected to any elements of the old and new deployment, including {CENTRAL} and {KIE_SERVERS}.
+. Create a temporary directory and change into this directory.
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-bc-proc.adoc:14]: *this* is confusing.

```suggestion
. Create a temporary directory and change into the new directory.
```

File: doc-content/enterprise-only/openshift/migration-ocp3-4-bc-proc.adoc [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+. Using the `oc` command, log in to the {OPENSHIFT} version 3 infrastructure and switch to the project containing the old deployment.
+. To view the pod names in the old deployment, run the command:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-bc-proc.adoc:16]:
```suggestion
. To view the pod names in the old deployment, run the following command:
```

File: doc-content/enterprise-only/openshift/migration-ocp3-4-bc-proc.adoc [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+. Use the `oc` command to copy the the `.niogit` repository and the Maven repository from the pod to the local machine, for example:
++
+[subs="attributes,verbatim,macros,specialchars,replacements"]
+----
+oc cp myapp-{PRODUCT_INIT}centr-5-689mw:/opt/kie/data/.niogit .niogit
+oc cp myapp-{PRODUCT_INIT}centr-5-689mw:/opt/kie/data/maven-repository maven-repository
+----
++
+. Using the `oc` command, log in to the {OPENSHIFT} version 4 infrastructure and switch to the project containing the new deployment.
+. To view the pod names in the new deployment, run the command:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-bc-proc.adoc:33]:
```suggestion
. To view the pod names in the new deployment, run the following command:
```

File: doc-content/enterprise-only/openshift/migration-ocp3-4-mysql-proc.adoc [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+.Prerequisites
+
+* You must have a machine that has network access to both the {OPENSHIFT} version 3 and {OPENSHIFT} version 4 infrastructures.
+* The `oc` command-line client from {OPENSHIFT} version 4 must be installed on the machine. For instructions about installing the command-line client, see https://access.redhat.com/documentation/en-us/openshift_container_platform/4.4/html/cli_tools/openshift-cli-oc#installing-the-cli[_CLI tools_] in {OPENSHIFT} documentation.
+* The `mysql` and `mysqldump` client applications provided by MySQL version 8 or later or by MariaDB version 10 or later must be installed.
+
+.Procedure
+
+. Ensure that no web clients and no client applications are connected to any elements of the old and new deployment, including {CENTRAL} and {KIE_SERVERS}.
+. Create a temporary directory and change into this directory.
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-mysql-proc.adoc:15]: Change into this directory? this sounds confusing. Are we changing to the new directory?

File: doc-content/enterprise-only/openshift/migration-ocp3-4-mysql-proc.adoc [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+. Using the `oc` command, log in to the {OPENSHIFT} version 3 infrastructure and switch to the project containing the old deployment.
+. To view the deployment configuration names in the old deployment, run the command:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-mysql-proc.adoc:17]:
```suggestion
. To view the deployment configuration names in the old deployment, run the following command:
```

File: doc-content/enterprise-only/openshift/migration-ocp3-4-mysql-proc.adoc [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+====
+You can also use the following commands to retrieve the user name and password:
+----
+oc get dc/myapp-postgresql -ojsonpath='{.spec.template.spec.containers[0].env[?(@.name=="POSTGRESQL_USER")]}'.value
+
+oc get dc/myapp-postgresql -ojsonpath='{.spec.template.spec.containers[0].env[?(@.name=="POSTGRESQL_PASSWORD")]}'.value
+----
+====
++
+. To view the service names in the old deployment, run the command:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-mysql-proc.adoc:54]:
```suggestion
. To view the service names in the old deployment, run the following command:
```

File: doc-content/enterprise-only/openshift/migration-ocp3-4-mysql-proc.adoc [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
++
+Find the `mysql` service that corresponds to the {KIE_SERVER}.
++
+. In a separate terminal window, start port forwarding from the local host to the `mysql` service, using the name and port number displayed for the service, for example:
++
+----
+oc port-forward service/myapp-mysql 3306:3306
+----
++
+. Create a full database dump using the recorded username, for example:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-mysql-proc.adoc:68]:
```suggestion
. Create a full database dump using the recorded user name, for example:
```

File: doc-content/enterprise-only/openshift/migration-ocp3-4-mysql-proc.adoc [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
++
+[subs="attributes,verbatim,macros,specialchars,replacements"]
+----
+mysqldump --all-databases -u {PRODUCT_INIT} -p -h 127.0.0.1 > mysqldump.sql
+----
++
+When prompted, enter the recorded password. The dump creation can take considerable time.
++ 
+. Stop the port forwarding in the separate window using the `Ctrl`+`C` key combination.
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-mysql-proc.adoc:77]: I'd suggest using keyboard shortcut syntax. 
```suggestion
. Stop the port forwarding in the separate window using the kbd:[Ctrl+C] key combination.
```

File: doc-content/enterprise-only/openshift/migration-ocp3-4-mysql-proc.adoc [hunk 13]
```diff
@@ -0,0 +1,129 @@
...
+. Using the `oc` command, log in to the {OPENSHIFT} version 4 infrastructure and switch to the project containing the new deployment.
+. To view the deployment configuration names in the new deployment, run the command:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-mysql-proc.adoc:79]:
```suggestion
. To view the deployment configuration names in the new deployment, run the following command:
```

File: doc-content/enterprise-only/openshift/migration-ocp3-4-mysql-proc.adoc [hunk 14]
```diff
@@ -0,0 +1,129 @@
...
+====
+You can also use the following commands to retrieve the user name and password:
+----
+oc get dc/myapp-postgresql -ojsonpath='{.spec.template.spec.containers[0].env[?(@.name=="POSTGRESQL_USER")]}'.value
+
+oc get dc/myapp-postgresql -ojsonpath='{.spec.template.spec.containers[0].env[?(@.name=="POSTGRESQL_PASSWORD")]}'.value
+----
+====
++
+. To view the service names in the new deployment, run the command:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-mysql-proc.adoc:106]:
```suggestion
. To view the service names in the new deployment, run the following command:
```

File: doc-content/enterprise-only/openshift/migration-ocp3-4-mysql-proc.adoc [hunk 15]
```diff
@@ -0,0 +1,129 @@
...
++
+Find the `mysql` service that corresponds to the {KIE_SERVER}.
++
+. In a separate terminal window, start port forwarding from the local host to the `mysql` service, using the name and port number displayed for the service, for example:
++
+----
+oc port-forward service/myappnew-mysql 3306:3306
+----
++
+. Restore the database dump using the recorded username, for example:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-mysql-proc.adoc:120]:
```suggestion
. Restore the database dump using the recorded user name, for example:
```

File: doc-content/enterprise-only/openshift/migration-ocp3-4-mysql-proc.adoc [hunk 16]
```diff
@@ -0,0 +1,129 @@
...
++
+[subs="attributes,verbatim,macros,specialchars,replacements"]
+----
+mysql -u {PRODUCT_INIT} -p -h 127.0.0.1 < mysqldump.sql
+----
++
+When prompted, enter the recorded password. The restoration can take considerable time.
++ 
+. Stop the port forwarding in the separate window using the `Ctrl`+`C` key combination.
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-mysql-proc.adoc:129]:
```suggestion
. Stop the port forwarding in the separate window using the kbd:[Ctrl+C] key combination.
```

File: doc-content/enterprise-only/openshift/migration-ocp3-4-postgresql-proc.adoc [hunk 17]
```diff
@@ -0,0 +1,138 @@
...
+.Prerequisites
+
+* You must have a machine that has network access to both the {OPENSHIFT} version 3 and {OPENSHIFT} version 4 infrastructures.
+* The `oc` command-line client from {OPENSHIFT} version 4 must be installed on the machine. For instructions about installing the command-line client, see https://access.redhat.com/documentation/en-us/openshift_container_platform/4.4/html/cli_tools/openshift-cli-oc#installing-the-cli[_CLI tools_] in {OPENSHIFT} documentation.
+* The `psql` and `pg_dump` client applications provided by PostgreSQL version 10 or later must be installed.
+
+.Procedure
+
+. Ensure that no web clients and no client applications are connected to any elements of the old and new deployment, including {CENTRAL} and {KIE_SERVERS}.
+. Create a temporary directory and change into this directory.
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-postgresql-proc.adoc:15]:
```suggestion
. Create a temporary directory and change into the new directory.
```

File: doc-content/enterprise-only/openshift/migration-ocp3-4-postgresql-proc.adoc [hunk 18]
```diff
@@ -0,0 +1,138 @@
...
+oc get dc/myapp-postgresql -ojsonpath='{.spec.template.spec.containers[0].env[?(@.name=="POSTGRESQL_USER")]}'.value
+
+oc get dc/myapp-postgresql -ojsonpath='{.spec.template.spec.containers[0].env[?(@.name=="POSTGRESQL_PASSWORD")]}'.value
+
+oc get dc/myapp-postgresql -ojsonpath='{.spec.template.spec.containers[0].env[?(@.name=="POSTGRESQL_DATABASE")]}'.value
+----
++
+====
++
+. To view the service names in the old deployment, run the command:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-postgresql-proc.adoc:59]:
```suggestion
. To view the service names in the old deployment, run the following command:
```

File: doc-content/enterprise-only/openshift/migration-ocp3-4-postgresql-proc.adoc [hunk 19]
```diff
@@ -0,0 +1,138 @@
...
+* You must have a machine that has network access to both the {OPENSHIFT} version 3 and {OPENSHIFT} version 4 infrastructures.
+* The `oc` command-line client from {OPENSHIFT} version 4 must be installed on the machine. For instructions about installing the command-line client, see https://access.redhat.com/documentation/en-us/openshift_container_platform/4.4/html/cli_tools/openshift-cli-oc#installing-the-cli[_CLI tools_] in {OPENSHIFT} documentation.
+* The `psql` and `pg_dump` client applications provided by PostgreSQL version 10 or later must be installed.
+
+.Procedure
+
+. Ensure that no web clients and no client applications are connected to any elements of the old and new deployment, including {CENTRAL} and {KIE_SERVERS}.
+. Create a temporary directory and change into this directory.
+. Using the `oc` command, log in to the {OPENSHIFT} version 3 infrastructure and switch to the project containing the old deployment.
+. To view the deployment configuration names in the old deployment, run the command:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-postgresql-proc.adoc:17]:
```suggestion
. To view the deployment configuration names in the old deployment, run the following command:
```

File: doc-content/enterprise-only/openshift/migration-ocp3-4-postgresql-proc.adoc [hunk 20]
```diff
@@ -0,0 +1,138 @@
...
++
+Find the `postgresql` service that corresponds to the {KIE_SERVER}.
++
+. In a separate terminal window, start port forwarding from the local host to the `postgresql` service, using the name and port number displayed for the service, for example:
++
+----
+oc port-forward service/myapp-postgresql 5432:5432
+----
++
+. Create a dump of the database using the recorded username and database name, for example:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-postgresql-proc.adoc:73]:
```suggestion
. Create a dump of the database using the recorded user name and database name, for example:
```

File: doc-content/enterprise-only/openshift/migration-ocp3-4-postgresql-proc.adoc [hunk 21]
```diff
@@ -0,0 +1,138 @@
...
++
+[subs="attributes,verbatim,macros,specialchars,replacements"]
+----
+pg_dump {PRODUCT_INIT}7 -h 127.0.0.1 -U {PRODUCT_INIT} -W >pgdump.sql
+----
++
+When prompted, enter the recorded password. The dump creation can take considerable time.
++ 
+. Stop the port forwarding in the separate window using the `Ctrl`+`C` key combination.
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-postgresql-proc.adoc:82]:
```suggestion
. Stop the port forwarding in the separate window using the kbd:[Ctrl+C] key combination.
```

File: doc-content/enterprise-only/openshift/migration-ocp3-4-postgresql-proc.adoc [hunk 22]
```diff
@@ -0,0 +1,138 @@
...
+. Using the `oc` command, log in to the {OPENSHIFT} version 4 infrastructure and switch to the project containing the new deployment.
+. To view the deployment configuration names in the new deployment, run the command:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-postgresql-proc.adoc:84]:
```suggestion
. To view the deployment configuration names in the new deployment, run the following command:
```

File: doc-content/enterprise-only/openshift/migration-ocp3-4-postgresql-proc.adoc [hunk 23]
```diff
@@ -0,0 +1,138 @@
...
++
+Find the `postgresql` service that corresponds to the {KIE_SERVER}.
++
+. In a separate terminal window, start port forwarding from the local host to the `postgresql` service, using the name and port number displayed for the service, for example:
++
+----
+oc port-forward service/myappnew-postgresql 5432:5432
+----
++
+. Restore the database dump using the recorded username and database name, for example:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-postgresql-proc.adoc:127]:
```suggestion
. Restore the database dump using the recorded user name and database name, for example:
```

File: doc-content/enterprise-only/openshift/migration-ocp3-4-postgresql-proc.adoc [hunk 24]
```diff
@@ -0,0 +1,138 @@
...
+[subs="attributes,verbatim,macros,specialchars,replacements"]
+----
+psql -h 127.0.0.1 -d {PRODUCT_INIT}7 -U {PRODUCT_INIT} -W <pgdump.sql 
+----
++
+When prompted, enter the recorded password. The restoration can take considerable time. 
++
+Review any displayed database error messages. Messages about objects that already exist are normal.
++ 
+. Stop the port forwarding in the separate window using the `Ctrl`+`C` key combination.
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/openshift/migration-ocp3-4-postgresql-proc.adoc:138]:
```suggestion
. Stop the port forwarding in the separate window using the kbd:[Ctrl+C] key combination.
```

**Pull request merged**